### DISCUSSION TITLE
How to add additional metrics to metrics-generator?

### DISCUSSION THREAD

**Discussion by Jht0309** [grafana/tempo#3661] [Q&A]
Excuse me
As the document shows, several metrics are generated automatically such as traces_spanmetrics_calls_total,etc. However, as for me, I need to add a new metric to count the download file size, which seems not to be assigned only by .yaml file.

I've tried to get it by:
1. using opentelemetry-otlp to generate metrics pipeline with the endpoint of  default tempo grpc(localhost:4317), then generate metrics at the end of the download function;
2. using opentelemetry-prometheus to generate metrics and send to prometheus endpoint;

but neither of them works, and then I added a label named "file_size“ which records the download file size to each span, and try to get the label value of file_size to calculate with metrics "traces_spanmetrics_calls_total" before visualization, but I can't retrieve the label value via PromQL,so it failed.

So how can I add my new metrics to span-metrics? Sincerely need your help T.T

**Comment by mapno** [comment]
Hi! Why can't you retrieve the value with PromQL? Is the label not being added to the metrics?

**Reply by Jht0309** [reply]
Hi,thanks for your reply!
In my code,I set the file download size as a label named "file_size" for each span, in order to record the file size downloaded this time by this url. Now I can filter metrics by "file_size" label and see the value of the label, thanks to span-metrics! 
But the problem is that I want to add an additional metrics generated by file_size, but I don't find any function about this. So I tried to use metrics "traces_spanmetrics_calls_total" and multiple the merics with the value of label "file_size" in order to calculate the total file size downloaded, but i'm failed as PromQL cannot use something like "label_values()" to get the value of label for calculation T.T 

I finally add the file_size_total metrics by directly using actix::web to run a "{ip:port}/metrics" interface for my server and opentelemetry_prometheus to expose the interface to prometheus docker, instead of trying to add additional metrics by adjusting span-metrics. So sincerely,is there any function to add additional metrics generated by data of trace/span in span-metrics? ^_^
